src/liquid-split.mapping.ts

```diff
@@ -392,27 +393,49 @@ function updateHolderOwnershipNonFactoryLiquidSplit(
   let liquidSplitId = liquidSplitAddress.toHexString()
 
   let fromAddressString = fromAddress.toHexString()
-  if (fromAddressString != ZERO_ADDRESS) {
-    let fromHolder = getHolder(
-      fromAddressString,
-      liquidSplitId,
-      blockNumber,
-      timestamp,
-    )
-    fromHolder.ownership = liquidSplitContract.scaledPercentBalanceOf(
-      fromAddress,
-    )
-    fromHolder.save()
-  }
-
   let toAddressString = toAddress.toHexString()
-  if (toAddressString != ZERO_ADDRESS) {
-    let toHolder = getHolder(
-      toAddressString,
-      liquidSplitId,
-      blockNumber,
-      timestamp,
-    )
+
+  if (fromAddressString == ZERO_ADDRESS || toAddressString == ZERO_ADDRESS) {
+    // If it's a mint or burn, need to update all holders ownership
+    if (fromAddressString != ZERO_ADDRESS) {
+      let fromHolder = getHolder(fromAddressString, liquidSplitId, blockNumber, timestamp)
+      let ownership = liquidSplitContract.scaledPercentBalanceOf(fromAddress)
+      if (ownership == ZERO) {
+        store.remove('Holder', fromHolder.id)
+      } else {
+        fromHolder.ownership = ownership
+        fromHolder.save()
+      }
+    }
+  
+    if (toAddressString != ZERO_ADDRESS) {
+      let toHolder = getHolder(toAddressString, liquidSplitId, blockNumber, timestamp)
+      toHolder.ownership = liquidSplitContract.scaledPercentBalanceOf(toAddress)
+      toHolder.save()
+    }
+
+    let liquidSplit = LiquidSplit.load(liquidSplitId) as LiquidSplit
+    let holders = liquidSplit.holders.load()
+    for (let i = 0; i < holders.length; i++) {
+      let holder = holders[i]
+      // Only update if it's not the from/to address (i.e. we haven't already updated it)
+      if (holder.account != fromAddressString && holder.account != toAddressString) {
+        holder.ownership = liquidSplitContract.scaledPercentBalanceOf(Address.fromString(holder.account))
+        holder.save()
+      }
+    }
+  } else {
+    // if it's a transfer, just update the from/to ownership
+    let fromHolder = getHolder(fromAddressString, liquidSplitId, blockNumber, timestamp)
+    let fromOwnership = liquidSplitContract.scaledPercentBalanceOf(fromAddress)
+    if (fromOwnership == ZERO) {
+      store.remove('Holder', fromHolder.id)
+    } else {
+      fromHolder.ownership = fromOwnership
+      fromHolder.save()
+    }
+  
+    let toHolder = getHolder(toAddressString, liquidSplitId, blockNumber, timestamp)
     toHolder.ownership = liquidSplitContract.scaledPercentBalanceOf(toAddress)
```

Comment:
can this be zero? if someone transfers nothing leading to a spam event we might still end up adding the holder